crates/fiber-lib/src/cch/trackers/lnd_trackers.rs

```diff
@@ -0,0 +1,537 @@
+    RouterClient, Uri,
+};
+use ractor::{Actor, ActorCell, ActorProcessingErr, ActorRef, OutputPort};
+use tokio::{select, time::sleep};
+use tokio_util::{sync::CancellationToken, task::TaskTracker};
+
+use crate::{
+    cch::{CchIncomingEvent, CchIncomingPaymentStatus, CchOutgoingPaymentStatus},
+    fiber::types::Hash256,
+};
+
+const MAX_CONCURRENT_INVOICE_TRACKERS: usize = 5;
+const INVOICE_TRACKING_TIMEOUT: Duration = Duration::from_secs(5 * 60); // 5 minutes
+
+/// LND connection information
+///
+/// This struct contains the connection details for communicating with an LND node.
+#[derive(Clone)]
+pub struct LndConnectionInfo {
+    pub uri: Uri,
+    pub cert: Option<Vec<u8>>,
+    pub macaroon: Option<Vec<u8>>,
+}
+
+impl LndConnectionInfo {
+    pub fn new(uri: Uri, cert: Option<Vec<u8>>, macaroon: Option<Vec<u8>>) -> Self {
+        Self {
+            uri,
+            cert,
+            macaroon,
+        }
+    }
+
+    pub async fn create_router_client(
+        &self,
+    ) -> Result<RouterClient, lnd_grpc_tonic_client::channel::Error> {
+        create_router_client(
+            self.uri.clone(),
+            self.cert.as_deref(),
+            self.macaroon.as_deref(),
+        )
+        .await
+    }
+
+    pub async fn create_invoices_client(
+        &self,
+    ) -> Result<InvoicesClient, lnd_grpc_tonic_client::channel::Error> {
+        create_invoices_client(
+            self.uri.clone(),
+            self.cert.as_deref(),
+            self.macaroon.as_deref(),
+        )
+        .await
+    }
+}
+
+/// Message types for the LndTrackerActor
+#[derive(Debug)]
+pub enum LndTrackerMessage {
+    /// Track a new invoice
+    TrackInvoice(Hash256),
+
+    /// Notification that an invoice tracker task has completed
+    ///
+    /// Sent by InvoiceTracker tasks when they terminate (either successfully
+    /// when invoice reaches final state, or due to error).
+    InvoiceTrackerCompleted {
+        payment_hash: Hash256,
+        completed_successfully: bool,
+    },
+
+    /// Get current state snapshot (for testing)
+    #[cfg(test)]
+    GetState(ractor::RpcReplyPort<StateSnapshot>),
+}
+
+/// Snapshot of actor state (for testing)
+#[cfg(test)]
+#[derive(Debug, Clone)]
+pub struct StateSnapshot {
+    pub invoice_queue_len: usize,
+    pub active_invoice_trackers: usize,
+}
+
+/// Arguments for starting the LndTrackerActor
+pub struct LndTrackerArgs {
+    pub port: Arc<OutputPort<CchIncomingEvent>>,
+    pub lnd_connection: LndConnectionInfo,
+    pub token: CancellationToken,
+    pub tracker: TaskTracker,
+}
+
+/// State for the LndTrackerActor
+pub struct LndTrackerState {
+    port: Arc<OutputPort<CchIncomingEvent>>,
+    lnd_connection: LndConnectionInfo,
+    token: CancellationToken,
+    tracker: TaskTracker,
+    /// Queue of payment hashes waiting to be tracked
+    invoice_queue: VecDeque<Hash256>,
+    /// Number of currently active invoice trackers
+    active_invoice_trackers: usize,
```

Comment:
I have found queue in the actor sometimes hard to debug, you may add metrics for monitoring.
```rust
gauge!("lnd_tracker.queue_length", state.invoice_queue.len() as f64);
gauge!("lnd_tracker.active_trackers", state.active_invoice_trackers as f64);

```